src/irc/ClientPool.ts

```diff
@@ -84,27 +77,22 @@ export class ClientPool {
         }
 
         // The client may not have signalled to us that it's connected, but it is connect*ing*.
-        const pending = Object.keys(this.virtualClients[server.domain].pending || {});
-        return pending.includes(nick);
+        return this.virtualClients[server.domain].pending.has(nick)
     }
 
     public killAllClients(): Bluebird<void[]> {
         const domainList = Object.keys(this.virtualClients);
         let clients: (BridgedClient|undefined)[] = [];
         domainList.forEach((domain) => {
             clients = clients.concat(
-                Object.keys(this.virtualClients[domain].nicks).map(
-                    (nick: string) => this.virtualClients[domain].nicks[nick]
-                )
+                [...this.virtualClients[domain].nicks.values()]
             );
 
             clients = clients.concat(
-                Object.keys(this.virtualClients[domain].userIds).map(
-                    (userId: string) => this.virtualClients[domain].userIds[userId]
-                )
+                [...this.virtualClients[domain].userIds.values()]
             );
 
-            clients.push(this.botClients[domain]);
+            clients.push(this.botClients.get(domain));
```

Comment:
why concat? and why re-wrap the output of `.values()` which creates a new array anyway?
```javascript
clients = [
  ...clients,
  ...this.virtualClients[domain].nicks.values(),
  ...this.virtualClients[domain].userIds.values(),
  this.botClients.get(domain),
];
```

Reply:
Good catch.